be/src/column/binary_column.cpp

```diff
@@ -66,24 +67,33 @@ void BinaryColumnBase<T>::append_selective(const Column& src, const uint32_t* in
     const auto& src_offsets = src_column.get_offset();
     const auto& src_bytes = src_column.get_bytes();
 
+    bool need_prefetch =
+            (size > SEQUENTIAL_DISTINCT * 2) && (src_offsets.back() > (1 << 22)) && src_offsets.size() > 64;
+
     size_t cur_row_count = _offsets.size() - 1;
     size_t cur_byte_size = _bytes.size();
 
     _offsets.resize(cur_row_count + size + 1);
+    std::vector<T> random_offsets(size);
     for (size_t i = 0; i < size; i++) {
+        if (need_prefetch && (SEQUENTIAL_DISTINCT + i < size)) {
```

Comment:
"Does each iterator trigger prefetching?

Reply:
i in [0, size-SEQUENTIAL_DISTINCT] will prefetch data (i+ SEQUENTIAL_DISTINCT)

Reply:
how to set a proper value for SEQUENTIAL_DISTINCT, is 16L based on experiments?

Reply:
@liuyehcf yes, previous dis in agg prefetch is also 16.